build.gradle

```diff
@@ -104,7 +104,7 @@ dependencies {
     compileOnly "org.apache.logging.log4j:log4j-slf4j-impl:2.22.0"
     compileOnly group: 'org.json', name: 'json', version: '20231013'
     zipArchive group: 'org.opensearch.plugin', name:'opensearch-sql-plugin', version: "${version}"
-    implementation("com.google.guava:guava:32.1.3-jre")
+    implementation("com.google.guava:guava:33.0.0-jre")
```

Comment:
Question about this update, `skills` depends on `ml-commons`, ml-common use different version of guava, it will cause jar hell exception when OpenSearch start

Reply:
In #38 it uses compileOnly, we should be consistent about this

Reply:
@dbwiddis can you help change this to compileOnly?